File: lib/domain/repository/book_repository.dart
import '../../core/model/book_metadata.dart';
import '../../core/model/book_search_result.dart';
import '../../core/model/opened_book.dart';
import '../../core/model/reading_progress.dart';

abstract class BookRepository {
  Stream<List<BookMetadata>> observeLibrary();
  Future<BookMetadata> importBook(String sourceUri);
  Future<OpenedBook> openBook(String bookId);
  Future<void> saveReadingProgress(ReadingProgress progress);
  Future<ReadingProgress?> getReadingProgress(String bookId);
  Future<List<BookSearchResult>> searchInBook(String bookId, String query);
}

File: domain/src/main/kotlin/com/readler/domain/repository/BookRepository.kt
package com.readler.domain.repository

import com.readler.core.model.BookMetadata
import com.readler.core.model.OpenedBook
import com.readler.core.model.BookSearchResult
import com.readler.core.model.ReadingProgress
import kotlinx.coroutines.flow.Flow

interface BookRepository {
    fun observeLibrary(): Flow<List<BookMetadata>>
    suspend fun importBook(sourceUri: String): BookMetadata
    suspend fun openBook(bookId: String): OpenedBook
    suspend fun saveReadingProgress(progress: ReadingProgress)
    suspend fun getReadingProgress(bookId: String): ReadingProgress?
    suspend fun searchInBook(bookId: String, query: String): List<BookSearchResult>
}
